cocos/bindings/dop/BufferAllocator.cpp

```diff
@@ -29,19 +29,21 @@
 
 namespace se {
 
-cc::map<PoolType, BufferAllocator *> BufferAllocator::_pools;
+cc::vector<BufferAllocator *> BufferAllocator::_pools(BUFFER_ALLOCATOR_SIZE);
 
 BufferAllocator::BufferAllocator(PoolType type)
 : _type(type) {
-    BufferAllocator::_pools[_type] = this;
+    if (IS_BUFFER_ALLOCATOR_TYPE(type)) {
+        BufferAllocator::_pools[GET_ARRAY_POOL_ID(type)] = this;
+    }
 }
 
 BufferAllocator::~BufferAllocator() {
     for (auto buffer : _buffers) {
         buffer.second->decRef();
     }
     _buffers.clear();
-    BufferAllocator::_pools.erase(_type);
+    _caches.clear();
```

Comment:
I think _buffers.clear() and  _caches.clear() is not needed in destructor.

Reply:
> I think _buffers.clear() and _caches.clear() is not needed in destructor.

Ok, I delete them.